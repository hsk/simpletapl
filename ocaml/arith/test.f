/* Examples for testing */

true;
if false then true else false; 

0; 
succ (pred 0);
iszero (pred (succ (succ 0))); 
iszero (pred (pred (succ (succ 0)))); 
iszero 0;

if 0 then succ(pred 0) else 0;
if 0 then succ(succ 0) else 0;
if 0 then succ(pred (succ 0)) else 0;
