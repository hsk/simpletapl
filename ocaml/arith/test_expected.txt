true
false
0
1
false
true
true
if 0 then succ (pred 0) else 0
if 0 then 2 else 0
if 0 then succ (pred 1) else 0
